### ISSUE TITLE
unclosed-quote's redirect-to-real-culprit fix only looks back one token, missing chains through comment-embedded quotes

### ISSUE THREAD

**Issue by walf443** [walf443/nginx-lint#299]
## Summary

PR #298 (fixing #295) redirects `UnclosedQuote`'s autofix to the real culprit line when a dangling artifact token is detected, by looking at the single most recently seen same-kind quoted-string token. This only handles a one-hop chain. A `"`/`'` character inside a trailing (end-of-line) `#` comment can create a longer chain that the current fix does not follow, reproducing the same wrong-line-fix bug #295 described.

## Repro

```
add_header X-Custom "value; # trailing comment with "quote" inside
    return 200 "ok";
```

Once the lexer is inside a double-quoted string (`add_header`'s), `#` has no special meaning — it's just string content. So the lexer's naive pairing consumes up to the *next* `"` character it finds, which is the one inside the comment's `"quote"`, not the end of the line. This produces:

- Token1: `"value; # trailing comment with "` — single line, ends with `"` (looks "closed"). This is actually the real culprit (this line has a missing closing quote that only *looks* closed because of the coincidental `"` inside the comment).
- Token2: `" inside\n    return 200 "` — spans two lines, ends with `"` (looks "closed"), but its own first line (`" inside`) has no semicolon.
- Token3: dangling, starting right after `ok`'s closing quote, running to EOF — this is what actually gets flagged.

PR #298's redirect only checks Token2 (the immediately preceding same-kind token) as the redirect target, and rejects it (no semicolon on Token2's own first line), so no redirect happens and the fix still corrupts `return`'s line (`ok"";`) exactly like before #298's fix — verified by running the real rule against this input.

## Why the current fix's heuristics don't extend cleanly

The natural fix would be "walk back through the whole chain of tokens that look closed only because they span multiple lines, not just the immediately preceding one." But in this repro, Token1 (the actual culprit) does **not** span multiple lines — it's entirely on line 1, since the comment content that swallowed its closing quote is on the same line. The assumption PR #298's fix makes ("a single-line-closed token is unrelated to any dangling-artifact chain, so the chain can't reach past it") does not hold here: a comment containing a stray quote character can make an otherwise-normal-looking single-line "closed" token be the actual problem.

A robust fix likely needs to reason about *comments* specifically (recognizing that a quote character following a `#` on the same source line, before the string's opening quote was reached, indicates the string swallowed what should have been comment text) rather than purely a length-of-chain heuristic over already-tokenized strings.

## Scope note

Filed separately per the maintainer's direction after PR #298 (a deliberately narrow fix for the reported #295 shape) was already in review — not blocking that PR.
